script.js

```diff
@@ -23,4 +21,69 @@ tags.forEach((tag) => {
   tag.addEventListener('click', () => {
     closeMenu();
   });
-});
\ No newline at end of file
+});
+
+const nameError = document.querySelector('#name-error');
+const emailError = document.querySelector('#email-error');
+const submitError = document.querySelector('#submit-error');
+function validate() {
+  const name = document.getElementById('user-name').value;
+  if (name.lenght === 0) {
+    nameError.innerHTML = 'Name is required';
+    return false;
+  }
+  if (!name.match(/^[A-Za-z]*\s{1}[A-Za-z]*$/)) {
+    nameError.innerHTML = 'Invalid name';
+    return false;
+  }
+  nameError.innerHTML = '<img src="checkbox-checked-solid-24 (1).png" alt="check icon">';
+  return true;
+}
+
+function validateEmail() {
+  const email = document.getElementById('user-email').value;
+
+  if (email.lenght === 0) {
+    emailError.innerHTML = 'Email is required';
+    return false;
+  }
+  if (!email.match(/^([a-za-z0-9_\-.])+@([a-za-z0-9_\-.])+\.([a-za-z]{2,4})$/)) {
+    emailError.innerHTML = 'Invalid email';
+    return false;
+  }
+  emailError.innerHTML = '<img src="checkbox-checked-solid-24 (1).png" alt="check icon">';
+  return true;
+}
+
+// LOCAL STORAGE
+const formName = document.getElementById('user-name');
+const email = document.getElementById('user-email');
+const formText = document.getElementById('user-text');
+
+const localData = () => {
+  const userData = {
+    userName: formName.value,
+    UserMail: email.value,
+    txt: formText.value,
+  };
+  localStorage.setItem('userData', JSON.stringify(userData));
+};
+const storedDetails = () => {
+  const info = JSON.parse(localStorage.getItem('userData'));
+  formName.value = info.formName;
+  email.value = info.email;
+  formText.value = info.formText;
+};
+
+function validateForm() {
+  if (!validate() || !validateEmail()) {
+    submitError.style.display = 'block';
+    submitError.innerHTML = 'Please fix errors to submit';
+    setTimeout(() => { submitError.style.display = 'none'; }, 3000);
+    return false;
+  }
+  localData();
+  storedDetails();
+  return true;
+}
+validateForm();
```

Comment:
- [ ] There is an issue in your code on line 80, the variable `submitError` returns `null`, and the `.style` method cannot be called on `null`. See the image below for a display of the error. 
- [ ] Also, notice that the red text invalid appears on page load which should not be. Kindly fix that

![image](https://github.com/Paulinelali/microverse-personal-portfolio/assets/84156714/3a092907-a84e-4c51-b815-ceaee45710f8)